src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataConstants.java

```diff
@@ -64,4 +64,87 @@ class HistoricalDataConstants {
 
   /** Tag value representing the NaN Float/Double value. */
   public static final String TAG_VALUE_NAN = "nan";
+
+  /**
+   * The name of the diagnostic tag that is populated with the number of seconds that the queue is
+   * running behind.
+   */
+  public static final String QUEUE_DIAGNOSTIC_TAG_RUNNING_BEHIND_SECONDS_NAME =
+      "ConnectorQueueBehindSeconds";
+
+  /**
+   * The description of the diagnostic tag that is populated with the number of seconds that the
+   * queue is running behind.
+   */
+  public static final String QUEUE_DIAGNOSTIC_TAG_RUNNING_BEHIND_SECONDS_DESC =
+      "Diagnostic tag containing the amount of time, in seconds, that the connector data queue is"
+          + " running behind.";
+
+  /**
+   * The type of the diagnostic tag (integer) that is populated with the number of seconds that the
+   * queue is running behind.
+   */
+  public static final int QUEUE_DIAGNOSTIC_TAG_RUNNING_BEHIND_SECONDS_TYPE = 3;
+
+  /**
+   * The default threshold for the number of seconds that the queue is running behind before the
+   * diagnostic tag value is displayed. This value is used when one is not explicitly provided.
+   *
+   * <p>The running behind time display threshold helps to prevent users from being alarmed by a
+   * running behind time which is within acceptable limits.
+   */
+  public static final int DEFAULT_QUEUE_DIAGNOSTIC_TAG_RUNNING_BEHIND_SECONDS_DISPLAY_THRESHOLD = 3;
```

Comment:
Three seconds does not appear to be enough time to avoid false positives considering that we have configurable polling times. The default que poll time for Thingworx is 3 seconds, so any additional delays beyond this would indicate a false positive.  
 I would guess that this needs to be more like  20 or 30 seconds to avoid false positives.

Reply:
I didn't realize that the check is after the EBD call. This makes the polling time less of an issue.

Reply:
I corrected this to `180`. The value of `3` was unrealistic, and I think I just made a copy/paste error or forgot to switch the value.

Realistically, connectors should be setting this threshold, so it shouldn't matter. For flexibility, I added a default value and the option to not specify though.